server/monitor/subscriptions.go

```diff
@@ -249,10 +265,19 @@ func (m *Monitor) recreateGlobalSubscription(subscriptionID, secret string) erro
 
 	m.metrics.ObserveSubscription(metrics.SubscriptionReconnected)
 
-	if err = m.store.DeleteSubscription(subscriptionID); err != nil {
-		m.api.LogDebug("Unable to delete old global subscription from DB", "subscriptionID", subscriptionID, "error", err.Error())
+	if subscriptionID == "" {
```

Comment:
Same concern as above: seems like we don't save the new subscription if `subscriptionID == ""`?

Reply:
Exact same situation, first is "client.DeleteSubscription", and later is `store.SaveGlobalSubscription`, so we are talking about different delete/save places.